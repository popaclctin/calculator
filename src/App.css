.calculator {
  font-size: 3rem;
  margin: 10rem auto;
  background-color: #a2790d;
}

.calculator button {
  display: block;
  width: 100%;
  font-size: 3rem;
  padding: 1rem 2rem;
  color: black;
  background-color: #ebd197;
  border: 0;
}

.calculator button:hover {
  background-color: #b48811;
  cursor: pointer;
}

.calculator button:active {
  background-color: white;
}

.result {
  background-color: #ffffff;
  color: black;
  height: 4rem;
  width: 10rem;
}

.result div {
  text-align: end;
  padding: 1rem 2rem;
  text-align: end;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}
