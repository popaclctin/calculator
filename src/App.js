import React, { Component } from 'react';
import './App.css';

class App extends Component {
  render() {
    return (
      <table className="calculator">
        <tr>
          <td colspan="4">
            <div className="result">0</div>
          </td>
        </tr>
        <tr>
          <td colspan="3">
            <Button label="clear" />
          </td>
          <td>
            <Button label="รท" />
          </td>
        </tr>
        <tr>
          <td>
            <Button label="7" />
          </td>
          <td>
            <Button label="8" />
          </td>
          <td>
            <Button label="9" />
          </td>
          <td>
            <Button label="ร" />
          </td>
        </tr>
        <tr>
          <td>
            <Button label="4" />
          </td>
          <td>
            <Button label="5" />
          </td>
          <td>
            <Button label="6" />
          </td>
          <td>
            <Button label="-" />
          </td>
        </tr>
        <tr>
          <td>
            <Button label="1" />
          </td>
          <td>
            <Button label="2" />
          </td>
          <td>
            <Button label="3" />
          </td>
          <td>
            <Button label="+" />
          </td>
        </tr>
        <tr>
          <td />
          <td>
            <Button label="0" />
          </td>
          <td colspan="2">
            <Button label="=" />
          </td>
        </tr>
      </table>
    );
  }
}

const Button = ({ label, onClick }) => (
  <button onClick={onClick}>{label}</button>
);

export default App;
