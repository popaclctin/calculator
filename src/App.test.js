import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders a snapshot', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should add correctly', () => {
  const component = renderer.create(<App />);
  component.getInstance().onClickNumber('7');
  component.getInstance().onClickOperation('ADD');
  component.getInstance().onClickNumber('4');
  component.getInstance().onEquals();
  expect(component.toJSON()).toMatchSnapshot();
});

it('should subtract correctly', () => {
  const component = renderer.create(<App />);
  component.getInstance().onClickNumber('7');
  component.getInstance().onClickOperation('SUBTRACT');
  component.getInstance().onClickNumber('4');
  component.getInstance().onEquals();
  expect(component.toJSON()).toMatchSnapshot();
});

it('should divide correctly', () => {
  const component = renderer.create(<App />);
  component.getInstance().onClickNumber('12');
  component.getInstance().onClickOperation('DIVIDE');
  component.getInstance().onClickNumber('4');
  component.getInstance().onEquals();
  expect(component.toJSON()).toMatchSnapshot();
});

it('should multiply correctly', () => {
  const component = renderer.create(<App />);
  component.getInstance().onClickNumber('7');
  component.getInstance().onClickOperation('MULTIPLY');
  component.getInstance().onClickNumber('4');
  component.getInstance().onEquals();
  expect(component.toJSON()).toMatchSnapshot();
});
