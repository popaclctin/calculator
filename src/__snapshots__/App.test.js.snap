// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<table
  className="calculator"
>
  <tbody>
    <tr>
      <td
        className="result"
        colSpan="4"
      >
        <div>
          0
        </div>
      </td>
    </tr>
    <tr>
      <td
        colSpan="3"
      >
        <button
          onClick={[Function]}
        >
          clear
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          ÷
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <button
          onClick={[Function]}
        >
          7
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          8
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          9
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          ×
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <button
          onClick={[Function]}
        >
          4
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          5
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          6
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          -
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <button
          onClick={[Function]}
        >
          1
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          2
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          3
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          +
        </button>
      </td>
    </tr>
    <tr>
      <td />
      <td>
        <button
          onClick={[Function]}
        >
          0
        </button>
      </td>
      <td
        colSpan="2"
      >
        <button
          onClick={[Function]}
        >
          =
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`should add correctly 1`] = `
<table
  className="calculator"
>
  <tbody>
    <tr>
      <td
        className="result"
        colSpan="4"
      >
        <div>
          11
        </div>
      </td>
    </tr>
    <tr>
      <td
        colSpan="3"
      >
        <button
          onClick={[Function]}
        >
          clear
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          ÷
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <button
          onClick={[Function]}
        >
          7
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          8
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          9
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          ×
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <button
          onClick={[Function]}
        >
          4
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          5
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          6
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          -
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <button
          onClick={[Function]}
        >
          1
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          2
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          3
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          +
        </button>
      </td>
    </tr>
    <tr>
      <td />
      <td>
        <button
          onClick={[Function]}
        >
          0
        </button>
      </td>
      <td
        colSpan="2"
      >
        <button
          onClick={[Function]}
        >
          =
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`should divide correctly 1`] = `
<table
  className="calculator"
>
  <tbody>
    <tr>
      <td
        className="result"
        colSpan="4"
      >
        <div>
          3
        </div>
      </td>
    </tr>
    <tr>
      <td
        colSpan="3"
      >
        <button
          onClick={[Function]}
        >
          clear
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          ÷
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <button
          onClick={[Function]}
        >
          7
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          8
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          9
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          ×
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <button
          onClick={[Function]}
        >
          4
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          5
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          6
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          -
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <button
          onClick={[Function]}
        >
          1
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          2
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          3
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          +
        </button>
      </td>
    </tr>
    <tr>
      <td />
      <td>
        <button
          onClick={[Function]}
        >
          0
        </button>
      </td>
      <td
        colSpan="2"
      >
        <button
          onClick={[Function]}
        >
          =
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`should multiply correctly 1`] = `
<table
  className="calculator"
>
  <tbody>
    <tr>
      <td
        className="result"
        colSpan="4"
      >
        <div>
          28
        </div>
      </td>
    </tr>
    <tr>
      <td
        colSpan="3"
      >
        <button
          onClick={[Function]}
        >
          clear
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          ÷
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <button
          onClick={[Function]}
        >
          7
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          8
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          9
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          ×
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <button
          onClick={[Function]}
        >
          4
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          5
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          6
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          -
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <button
          onClick={[Function]}
        >
          1
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          2
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          3
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          +
        </button>
      </td>
    </tr>
    <tr>
      <td />
      <td>
        <button
          onClick={[Function]}
        >
          0
        </button>
      </td>
      <td
        colSpan="2"
      >
        <button
          onClick={[Function]}
        >
          =
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`should subtract correctly 1`] = `
<table
  className="calculator"
>
  <tbody>
    <tr>
      <td
        className="result"
        colSpan="4"
      >
        <div>
          3
        </div>
      </td>
    </tr>
    <tr>
      <td
        colSpan="3"
      >
        <button
          onClick={[Function]}
        >
          clear
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          ÷
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <button
          onClick={[Function]}
        >
          7
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          8
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          9
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          ×
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <button
          onClick={[Function]}
        >
          4
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          5
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          6
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          -
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <button
          onClick={[Function]}
        >
          1
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          2
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          3
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          +
        </button>
      </td>
    </tr>
    <tr>
      <td />
      <td>
        <button
          onClick={[Function]}
        >
          0
        </button>
      </td>
      <td
        colSpan="2"
      >
        <button
          onClick={[Function]}
        >
          =
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;
